pandaserver/daemons/scripts/add.py

```diff
@@ -0,0 +1,560 @@
+                file = fileList.pop(0)
+                # check format
+                match = re.search('^(\d+)_([^_]+)_.{36}(_\d+)*$',file)
+                if match is not None:
+                    fileName = '%s/%s' % (dirName,file)
+                    if not os.path.exists(fileName):
+                        continue
+                    try:
+                        modTime = datetime.datetime(*(time.gmtime(os.path.getmtime(fileName))[:7]))
+                        thr = None
+                        if (timeNow - modTime) > datetime.timedelta(hours=24):
+                            # last chance
+                            tmpLog.debug("Last Add File {0} : {1}".format(os.getpid(),fileName))
+                            thr = AdderGen(taskBuffer,match.group(1),match.group(2),fileName,
+                                           ignoreTmpError=False,siteMapper=aSiteMapper)
+                        elif (timeInt - modTime) > datetime.timedelta(minutes=gracePeriod):
+                            # add
+                            tmpLog.debug("Add File {0} : {1}".format(os.getpid(),fileName))
+                            thr = AdderGen(taskBuffer,match.group(1),match.group(2),fileName,
+                                           ignoreTmpError=True,siteMapper=aSiteMapper)
+                        if thr is not None:
+                            thr.run()
+                    except Exception:
+                        type, value, traceBack = sys.exc_info()
+                        tmpLog.error("%s %s" % (type,value))
+
+
+        # launcher
+        def launch(self,taskBuffer,aSiteMapper,holdingAna):
+            # run
+            self.process = multiprocessing.Process(target=self.run,
+                                                   args=(taskBuffer,aSiteMapper,holdingAna))
+            self.process.start()
+
+
+        # join
+        def join(self):
+            self.process.join()
+
+
+
+    # get buildJobs in the holding state
+    holdingAna = []
+    varMap = {}
+    varMap[':prodSourceLabel'] = 'panda'
+    varMap[':jobStatus'] = 'holding'
+    status,res = taskBuffer.querySQLS("SELECT PandaID from ATLAS_PANDA.jobsActive4 WHERE prodSourceLabel=:prodSourceLabel AND jobStatus=:jobStatus",varMap)
+    if res is not None:
+        for id, in res:
+            holdingAna.append(id)
+    tmpLog.debug("holding Ana %s " % holdingAna)
+
+    # add files
+    tmpLog.debug("Adder session")
+
+    # make TaskBuffer IF
+    taskBufferIF = TaskBufferInterface()
+    taskBufferIF.launch(taskBuffer)
+
+    adderThrList = []
+    for i in range(3):
+        p = AdderProcess()
+        p.launch(taskBufferIF.getInterface(),aSiteMapper,holdingAna)
+        adderThrList.append(p)
+
+    # join all threads
+    for thr in adderThrList:
+        thr.join()
+
+    # join sender
+    mailSender.join()
+
+    # join fork threads
+    for thr in forkThrList:
+        thr.join()
+
+    # terminate TaskBuffer IF
+    taskBufferIF.terminate()
```

Comment:
@tmaeno 

I got a problem that the signal handler (`signal.signal`) in the daemon master process somehow can catch the SIGTERM from this line in add.py
```
taskBufferIF.terminate()
```
which is running in a worker (child) process. I've got no idea why.
So now the whole daemon gets killed after some worker runs "add.py"...

It can be "solved" by commenting out this line, but not elegant...

Do you have better idea? Thanks.